Ghidra/Processors/V850E2M/data/languages/Instructions/Float.sinc

```diff
@@ -0,0 +1,474 @@
+##################################################
+#####				  Float					 #####
+##################################################
+
+
+# ABSF.D reg2, reg3 - rrrr011111100000|wwww010001011000
+:absf.d R1115x2, R2731x2 is R1115x2 & op0510=0x3F & op0004=0b00000; R2731x2 & op2126=0b100010 & op1620=0b11000
+{
+	R2731x2 = abs(R1115x2);
+}
+
+# ABSF.S reg2, reg3 - rrrrr11111100000|wwwww10001001000
+:absf.s R1115, R2731 is R1115 & op0510=0x3F & op0004=0b00000; R2731 & op2126=0b100010 & op1620=0b01000
+{
+	R2731 = abs(R1115);
+}
+
+# ADDF.D reg1, reg2, reg3 - rrrr0111111RRRR0|wwww010001110000
+:addf.d R0004x2, R1115x2, R2731x2 is R1115x2 & op0510=0x3F & R0004x2 ; R2731x2 & op2126=0b100011 & op1620=0b10000
+{
+	R2731x2 = R1115x2 f+ R0004x2;
+}
+
+# ADDF.S reg1, reg2, reg3 - rrrrr111111RRRRR|wwwww10001100000
+:addf.s R0004, R1115, R2731 is R1115 & op0510=0x3F & R0004 ; R2731 & op2126=0b100011 & op1620=0b00000
+{
+	R2731 = R1115 f+ R0004;
+}
+
+# CEILF.DL reg2, reg3 - rrrr011111100010|wwww010001010100
+:ceilf.dl R1115x2, R2731x2 is R1115x2 & op0510=0x3F & op0004=0b00010; R2731x2 & op2126=0b100010 & op1620=0b10100
+{
+	local var:8 = ceil(float2float(R1115x2));
+	R2731x2 = trunc(var);
+}
+
+# CEILF.DUL reg2, reg3 - rrrr011111110010|wwww010001010100
+:ceilf.dul R1115x2, R2731x2 is R1115x2 & op0510=0x3F & op0004=0b10010; R2731x2 & op2126=0b100010 & op1620=0b10100
+{
+	local var:8 = ceil(float2float(R1115x2));
+	R2731x2 = trunc(var);
+}
+
+# CEILF.DUW reg2, reg3 - rrrrr11111110010|wwwww10001010000
+:ceilf.duw R1115x2, R2731 is R1115x2 & op0510=0x3F & op0004=0b10010; R2731 & op2126=0b100010 & op1620=0b10000
+{
+	R2731 = trunc(ceil(R1115x2));
+}
+
+# CEILF.DW reg2, reg3 - rrrrr11111100010|wwwww10001010000
+:ceilf.dw R1115x2, R2731 is R1115x2 & op0510=0x3F & op0004=0b00010; R2731 & op2126=0b100010 & op1620=0b10000
+{
+	R2731 = trunc(ceil(R1115x2));
+}
+
+# CEILF.SL reg2, reg3 - rrrrr11111100010|wwww010001000100
+:ceilf.sl R1115, R2731x2 is R1115 & op0510=0x3F & op0004=0b00010; R2731x2 & op2126=0b100010 & op1620=0b00100
+{
+	local var:8 = ceil(float2float(R1115));
```

Comment:
I'm not great at float code in SLEIGH, but is the `float2float` needed here?, i think its used to move between precision, but it looks like you're starting with 8-byte and ending in 8-byte

Reply:
current code is correct, because `R1115` (and any other register without `x2` suffix) is 32-bits wide

Reply:
Oops, thanks @esaulenka, I commented on wrong line. `ceilf.dul` and `ceilf.dl` for example though